src/backend/distributed/transaction/backend_data.c

```diff
@@ -1130,3 +1223,72 @@ DecrementClientBackendCounter(void)
 {
 	pg_atomic_sub_fetch_u32(&backendManagementShmemData->activeClientBackendCounter, 1);
 }
+
+
+/*
+ * CitusSignalBackend gets a global pid and and ends the original query with the global pid
+ * that might have started in another node by connecting to that node and running either
+ * pg_cancel_backend or pg_terminate_backend based on the withTerminate argument.
+ */
+static bool
+CitusSignalBackend(uint64 globalPID, uint64 timeout, int sig)
+{
+	Assert((sig == SIGINT) || (sig == SIGTERM));
+
+	int nodeId = ExtractNodeIdFromGlobalPID(globalPID);
```

Comment:
If the nodeId is `GLOBAL_PID_NODE_ID_FOR_NODES_NOT_IN_METADATA`, maybe we give a nicer error? 
ERROR: The global_pid (%s) you canceled/terminated maps to a node that does not exists in Citus' metadata.
HINT: Connect to the node directly and run pg_cancel/terminate_backend(pid);